bff-service/index.js

```diff
@@ -0,0 +1,64 @@
+const express = require('express');
+const axios = require('axios').default;
+require('dotenv').config();
+
+const app = express();
+const PORT = process.env.PORT || 3001;
+let products = {
+  data: null,
+  timestamp: null,
+  valid: function() {
+    return Date.now() - this.timestamp < 120000;
+  }
+};
+
+app.use(express.json());
+
+app.all('/*', (req, res) => {
+  console.log('originalUrl', req.originalUrl);
+  console.log('method', req.method);
+  console.log('body', req.body);
+
+  const recipient = req.originalUrl.split('/')[1];
+  console.log('recipient', recipient);
+
+  const recipientURL = process.env[recipient];
+  console.log('recipientURL', recipientURL);
+  console.log('cache:', products.valid()); 
+
+  if (recipient === 'products' && products.data && products.valid()) {
+    res.status(200).json(products.data);
+  } else if (recipientURL) {
+    const axiosConfig = {
+      method: req.method,
+      url: `${recipientURL}${req.originalUrl}`,
+      ...(Object.keys(req.body || {}).length > 0 && {data: req.body})
+    };
+
+    console.log('axiosConfig', axiosConfig);
+
+    axios(axiosConfig)
+    .then(function(response) {
```

Comment:
Not an error, but you can use arrow function to be consistent.

Reply:
thank you! will fix this